addons/l10n_ec/models/res_partner.py

```diff
@@ -0,0 +1,98 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields, api
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+def verify_final_consumer(vat):
+    b = True
+    c = 0
+    try:
+        for n in vat:
+            if int(n) != 9:
+                b = False
+            c += 1
+        if c == 13:
+            return b
+    except Exception as e:
+        _logger.debug("Error parsing final customer value %s", e)
+        return False
+
+
+class ResPartner(models.Model):
+
+    _inherit = "res.partner"
+
+    def check_vat_ec(self, vat):
+        if self.l10n_latam_identification_type_id.is_vat:
+            consumidor_final = verify_final_consumer(vat)
+            if consumidor_final:
+                return consumidor_final, "Consumidor"
+            if self.l10n_latam_identification_type_id.name == "Ced":
+                return self.is_valid_ci_ec(vat), "Cedula"
+            elif (
+                self.l10n_latam_identification_type_id.name == "RUC"
+                and vat != "9999999999999"
+            ):
+                return self.is_valid_ruc_ec(vat), "Ruc"
+        return True, False
+
+    l10n_ec_type_sri = fields.Char(
+        "SRI Identification Type",
+        store=True,
+        readonly=True,
+        compute="_compute_l10n_ec_type_sri",
+    )
+
+    @api.depends("vat", "country_id", "l10n_latam_identification_type_id")
+    def _compute_l10n_ec_type_sri(self):
+        passport_ref = self.env.ref("l10n_ec.ec_passport", False)
+        for partner in self:
+            vat_type = ""
+            if partner.country_id:
+                if partner.vat and partner.country_id.code == "EC":
+                    try:
+                        _dni, vat_type = self.check_vat_ec(partner.vat)
+                    except Exception as e:
+                        _logger.debug("Error checking vat, error %s", e)
+                if partner.country_id.code != "EC" or (
+                    passport_ref
+                    and partner.country_id.code == "EC"
+                    and partner.l10n_latam_identification_type_id.id == passport_ref.id
+                ):
+                    vat_type = "Pasaporte"
+            partner.l10n_ec_type_sri = vat_type
+
+    l10n_ec_foreign = fields.Boolean(
+        "Foreign?",
+        readonly=True,
+        store=True,
+        compute="_compute_l10n_ec_foreign",
+    )
+    l10n_ec_foreign_type = fields.Selection(
+        [
+            ("01", "Persona Natural"),
```

Comment:
should be based on vat or not

Reply:
**Ideas**:
country of the partner_id is not EC
Add a constraint to check Identification type > on invoice validation (when company_id = "EC")

Reply:
Removed, python implementation calc